azure-kusto-data/azure/kusto/data/helpers.py

```diff
@@ -50,9 +52,17 @@ def dataframe_from_result_table(table: "Union[KustoResultTable, KustoStreamingRe
     for col in table.columns:
         if col.column_type == "bool":
             frame[col.column_name] = frame[col.column_name].astype(bool)
-        if col.column_type == "datetime":
+        elif col.column_type == "int" or col.column_type == "long":
+            frame[col.column_name] = frame[col.column_name].astype("Int64")
+        elif col.column_type == "real" or col.column_type == "decimal":
+            frame[col.column_name] = frame[col.column_name].replace('NaN', np.NaN)
+            frame[col.column_name] = frame[col.column_name].replace('Infinity', np.PINF)
+            frame[col.column_name] = frame[col.column_name].replace('-Infinity', np.NINF)
+            frame[col.column_name] = pd.to_numeric(frame[col.column_name], errors= 'coerce')
```

Comment:
I think the tests are failing because of a formatting issue here. The space after errors= should be removed.
```suggestion
            frame[col.column_name] = pd.to_numeric(frame[col.column_name], errors='coerce')
```